doc/us-en/tutorials/Device-connect-to-OpenEdge-with-hub-module.md

```diff
@@ -1,3 +1,149 @@
+        permit: ['#']
+logger:
+  path: var/log/openedge/service.log
+  level: "debug"
+```
+
+As mentioned above, when the Hub Module starts, it will open ports 1883, 8883 and 8080 at the same time, which used them as TCP, SSL, WS (Websocket) connection ports. And we will use MQTTBOX and MQTT.fx as MQTT Client to connect to OpenEdge . 
+
+**TCP Connection Test**
+
+Start the MQTT.fx, enter the Edit Connection Profiles page, fill in the Profile Name, fill in the Broker Address and Port according to the connection configuration of the OpenEdge Hub module, and then configure the username & password in User Credentials according to the `principals` configuration option. Then click `Apply` button to complete the connection configuration of MQTT.fx with TCP protocal as shown below.
+
+![TCP connection configuration](../../images/tutorials/connect/mqttbox-tcp-connect-config.png)
+
+Then close the configuration page, select the Profile Name configured, then click `Connect` button, if the connection configuration information matches the permission information allowed in the `principals` configuration item of OpenEdge Hub module, you can see the connection success flag, as shown below .
+
+![TCP connection success](../../images/tutorials/connect/mqttbox-tcp-connect-success.png)
+
+**SSL Connection Test**
+
+Start the MQTT.fx and enter the Edit Connection Profiles page. Similar to the TCP connection configuration, fill in the corresponding Profile Name, Broker Address, and Broker Port. For SSL protocal, you need to fill in the User Name in User Credentials and configure SSL/TLS option as shown below. Then click the `Apply` button to complete the connection configuration of MQTT.fx in SSL connection mode.
+
+![SSL connection configuration1](../../images/tutorials/connect/mqttbox-ssl-connect-config1.png)
+
+![SSL connection configuration2](../../images/tutorials/connect/mqttbox-ssl-connect-config2.png)
+
+Then close the configuration page, select the Profile Name configured, then click `Connect` button, if the connection configuration information matches the permission information allowed in the `principals` configuration item of OpenEdge Hub module, you can see the connection success flag, as shown below .
+
+![SSL connection success](../../images/tutorials/connect/mqttbox-ssl-connect-success.png)
+
+**WS (Websocket) Connection Test**
+
+Start the MQTTBOX, enter the Client creation page, select the `ws` protocol, configure the address and port according to the OpenEdge Hub module, and configure the username and password according to the `principals` configuration item, then click the save button. Complete the connection configuration of MQTTBOX in WS connection mode, as shown in the figure below.
+
+![WS（Websocket）connection configuration](../../images/tutorials/connect/mqttbox-ws-connect-config.png)
+
+As long as the above operation is correct, you can see the sign of successful connection with OpenEdge Hub in MQTTBOX, as shown in the figure below.
```

Comment:
Once the above operation is correct, you can see the sign of successful connection with OpenEdge Hub in MQTTBOX, which is shown in the figure as below.